docs/docs-beta/docs/guides/report-events-to-datadog.md

```diff
@@ -0,0 +1,210 @@
+---
+title: Log Dagster events to Datadog
+description: Learn how to report Dagster events to Datadog
+sidebar_label: Log Events to Datadog
+---
+
+It's possible to monitor the events that occur in Dagster through Datadog by implementing a custom sensor; a mapping occurs between the Dagster event, and then the event is logged.
+
+<details>
+<summary>Prerequisites</summary>
+
+To follow this guide, you'll need:
+
+- Familiarity with [Assets](/concepts/sensors)
+</details>
+
+## Creating a custom sensor
+
+First, we create a method that polls the Dagster instance for all events that have occurred since the last invocation of the sensor. The method `poll_instance_for_latest_events` is defined which takes in a `DagsterInstance`, the previous `cursor` value, and additional arguments for filtering of events.
+
+
+```python title="poll_instance.py"
+from typing import List
+
+from dagster import (
+    AssetKey,
+    DagsterEventType,
+    DagsterInstance,
+    EventLogRecord,
+    EventRecordsFilter,
+)
+
+DEFAULT_MONITORED_EVENT_TYPES = [
+    DagsterEventType.ASSET_MATERIALIZATION,
+    DagsterEventType.STEP_FAILURE,
+    DagsterEventType.STEP_SUCCESS,
+    DagsterEventType.STEP_SKIPPED,
+    DagsterEventType.PIPELINE_SUCCESS,
+    DagsterEventType.PIPELINE_FAILURE,
+    DagsterEventType.ASSET_CHECK_EVALUATION,
+]
+
+
+def poll_instance_for_latest_events(
+    instance: DagsterInstance,
+    cursor: int | None = None,
+    asset_key: AssetKey | None = None,
+    event_types: List[DagsterEventType] = DEFAULT_MONITORED_EVENT_TYPES,
+    limit: int = 500,
+) -> List[EventLogRecord]:
+    "Takes a Dagster instance and fetches the latest records from it."
+    all_events: List[EventLogRecord] = []
+    for t in event_types:
+        matched = instance.get_event_records(
```

Comment:
We're trying to deprecate this generic `get_event_records` call, in favor of more narrowly scoped fetch methods (e.g. `fetch_materializations` / `fetch_run_status_changes` / `get_asset_check_execution_history`).  

The event types (in `DEFAULT_MONITORED_EVENT_TYPES`) not covered are the step events, which we need to find a way to instrument from inside the run worker.

Reply:
Thanks. I'll go ahead and mark this as a draft until further progress has been made on the fetch methods.